### PR TITLE
Updates to util code, info.c

### PR DESCRIPTION
**Pull request by a-ilango**
- Update info.c to support all modes
- Correct match function for util_find_domain
- Store mr_mode bits with util_domain

### CODE REVIEW

File: prov/util/src/util_domain.c [hunk 1]
```diff
@@ -64,8 +64,9 @@ static int util_domain_init(struct util_domain *domain,
 {
 	atomic_initialize(&domain->ref, 0);
 	fastlock_init(&domain->lock);
-	domain->caps = info->caps;
-	domain->mode = info->mode;
+	domain->caps = info->caps | info->domain_attr->caps;
+	domain->mode = info->mode | info->domain_attr->mode;
```
Comment (shefty) [inline prov/util/src/util_domain.c]: Now that we have domain specific mode/caps, I think setting domain->caps/mode will lead to confusion later.  Maybe rename these to all_caps, info_caps, merged_caps, naming_is_hard_caps to better indicate that these are not the domain specific flags.

Comment (a-ilango) [inline prov/util/src/util_domain.c]: How about info_domain_caps/mode?

**Comment by shefty** [comment]
Sounds good

**Comment by a-ilango** [comment]
Updated the PR.

**Pull request merged**